vllm/executor/ray_gpu_executor.py

```diff
@@ -221,9 +236,23 @@ def _driver_execute_model(
         Passing None will cause the driver to stop the model execution
         loop running in each of the remote workers.
         """
+        assert not USE_SPMD_WORKER, (
+            "driver_worker does not exist for VLLM_USE_SPMD_WORKER=1")
         return self.driver_worker.execute_method("execute_model",
                                                  execute_model_req)
 
+    def execute_model(
+            self,
+            execute_model_req: ExecuteModelRequest) -> List[SamplerOutput]:
+        if not USE_SPMD_WORKER:
+            return super().execute_model(execute_model_req)
+
+        if self.forward_dag is None:
+            self.forward_dag = self._compiled_ray_dag(enable_asyncio=False)
+
+        outputs = ray.get(self.forward_dag.execute(execute_model_req))
+        return outputs[0]
+
     def _run_workers(
```

Comment:
Is it correct if adag is used, it is not used to execute model? Maybe we should comment it.

Reply:
The workers of ADAG executes the model, here we pass `execute_model_req` to the ADAG (therefore its workers) so that the model will be executed by the workers.